### PR TITLE
[stable/prometheus-adapter] Add support for image pull secrets

### PR DESCRIPTION
**Pull request by ivanmp91**
Added support to pull custom metrics image from private registry using k8s secrets for registry authentication.

### CODE REVIEW

**Comment by paulczar** [comment]
please sign your commits (dco) and fix the conflicts

**Comment by ivanmp91** [comment]
> please sign your commits (dco) and fix the conflicts

Done!

**Comment by cpanato** [comment]
@ivanmp91 please sign the DCO

File: stable/prometheus-adapter/Chart.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 name: prometheus-adapter
-version: v0.3.0
+version: v0.3.1
```
Comment (cpanato) [inline stable/prometheus-adapter/Chart.yaml]: bump to 0.4.0

**Comment by ivanmp91** [comment]
@cpanato Thanks! I've done all the changes requested. Please let me know if you need anything else!

**Pull request merged**